dexmaker-tests/src/androidTest/java/com/android/dx/AnnotationIdTest.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.android.dx;
+
+import android.support.test.InstrumentationRegistry;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.lang.annotation.*;
+import java.lang.reflect.Method;
+
+import static java.lang.reflect.Modifier.PUBLIC;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public final class AnnotationIdTest {
+
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target({ElementType.METHOD})
+    @interface MethodAnnotation {
+        ElementEnum elementEnum() default ElementEnum.INSTANCE_0;
+        boolean elementBoolean() default false;
+        byte elementByte() default Byte.MIN_VALUE;
+        char elementChar() default 'a';
+        double elementDouble() default Double.MIN_NORMAL;
+        float elementFloat() default Float.MIN_NORMAL;
+        int elementInt() default Integer.MIN_VALUE;
+        long elementLong() default Long.MIN_VALUE;
+        short elementShort() default Short.MIN_VALUE;
+        String elementString() default "foo";
+        Class<?> elementClass() default Object.class;
+    }
+
+    enum ElementEnum {
+        INSTANCE_0,
+        INSTANCE_1,
+    }
+
+    private DexMaker dexMaker;
+    private static TypeId<?> GENERATED = TypeId.get("LGenerated;");
+
+    @Before
+    public void setUp() throws Exception {
+        reset();
+    }
+
+    @Test
+    public void testAddToMethod() throws Exception {
```

Comment:
Can we break this up into smaller / more focused tests? I'm imagining if this one monolith test starts failing, it'll be quite a pain to debug what happened since each section is just adding on to the same generated file.

I also like to make the test method names a little more descriptive about what they're doing. Since we're using JUnit 4, we don't need to prefix the method name with `test` and we can make it a more valuable (e.g. `addBooleanAnnotationToMethod`, etc)

Reply:
Yes, I will break it up.

Reply:
It's done by new commit 733f075